libevmasm/Assembly.cpp

```diff
@@ -629,15 +629,18 @@ LinkerObject const& Assembly::assemble() const
 			ret.bytecode.resize(ret.bytecode.size() + bytesPerDataRef);
 			break;
 		case PushSub:
-			assertThrow(i.data() <= numeric_limits<size_t>::max(), AssemblyException, "");
 			ret.bytecode.push_back(dataRefPush);
-			subRef.insert(make_pair(static_cast<size_t>(i.data()), ret.bytecode.size()));
+			subRef.insert(make_pair(i.data(), ret.bytecode.size()));
 			ret.bytecode.resize(ret.bytecode.size() + bytesPerDataRef);
 			break;
 		case PushSubSize:
 		{
-			assertThrow(i.data() <= numeric_limits<size_t>::max(), AssemblyException, "");
-			auto s = m_subs.at(static_cast<size_t>(i.data()))->assemble().bytecode.size();
+			vector<size_t> subIds = decodeSubIds(i.data());
+			Assembly const* currentAssembly = this;
+			for (size_t currentSubId: subIds)
+				currentAssembly = currentAssembly->m_subs.at(currentSubId).get();
+			assertThrow(currentAssembly, AssemblyException, "");
```

Comment:
I'd put the assert inside the loop:
```suggestion
			for (size_t currentSubId: subIds)
			{
				assertThrow(currentAssembly, AssemblyException, "");
				currentAssembly = currentAssembly->m_subs.at(currentSubId).get();
			}
```

Also, maybe it would be a good idea to create a helper function for getting a subassembly by encoded ID. It would not be used much but at least it would make this code simpler.

Reply:
I'm wondering if a private function in `Assembly` that just calls that function again on the sub-assembly would be a simpler way to implement this.